### DISCUSSION TITLE
Support backteating with two different timeframes

### DISCUSSION THREAD

**Discussion by ramilexe** [rodrigo-brito/ninjabot#248] [Q&A] [answered]
I would like to run backteating for BTCUSDT pair 1h timeframe but then simulate exchange behavior using 1s timeframe to have more realistic analys. Is it possible to do using existing code or it is a new feature?

**Comment by rodrigo-brito** [comment, marked as answer]
Hi @ramilexe, I never tested with `1s` timeframe. But it works fine with `1m` and above. I believe that Binance do not provide the 1s timeframe data feed to download. Because the real time feed is updated each 2s.

To use this feature, you have to load a small time frame in the backtest, example:

```go
	csvFeed, err := exchange.NewCSVFeed(
		strategy.Timeframe(),
		exchange.PairFeed{
			Pair:      "BTCUSDT",
			File:      "testdata/btc-1m.csv",
			Timeframe: "1m",
		},
		exchange.PairFeed{
			Pair:      "ETHUSDT",
			File:      "testdata/eth-1m.csv",
			Timeframe: "1h",
		},
	)
```

And in your strategy you have to specify a timeframe bigger than the dataset, e.g.: 1h

**Comment by ramilexe** [comment]
Thank you, does the order of `PairFeed` matter?

**Comment by rodrigo-brito** [comment]
I think it will not affect the trading results.

**Comment by adamantmm** [comment]
For realistic fills you usually want decision TF and execution TF split: signal on 1h, simulate fills against a finer series (1m is the practical floor on most Binance historical endpoints — 1s historical klines are not available the way the live stream is).

Pattern that works with the current PairFeed setup:

1. Strategy runs on the slow feed and only emits intent.
2. A thin execution layer consumes the fast feed and decides whether the limit/stop would have been touched intra-bar.

Watch for double-counting if both feeds call `OnCandle` on the same strategy instance — keep indicators on the slow series and use the fast series read-only for fill checks. Order of PairFeed registration mostly does not matter for results; what matters is not letting the fast feed mutate your signal state.
